### DISCUSSION TITLE
Restarting MALAMUTE Simulation

### DISCUSSION THREAD

**Discussion by raj-anant** [idaholab/moose#24347] [Q&A General]
Dear MOOSE Team,

I am trying to resume a MOOSE simulation using the results of a previous simulation as the initial state. I am following the strategy given in the following example:

https://mooseframework.inl.gov/application_usage/restart_recover.html

I read in the mesh from the previous result:
```
[Mesh]
    file = previous.e
[]
```
I am running a MLAMUTE simulation and need to read the temperature and velocity profiles from the previous results. Temperature is a scalar, and so works easily using the following:
```
[temp]
    initial_from_file_var = temp
  []
```
However, velocity is a vector, and the results save the velocity as veloctiy_x and velocity_y. How can I read the two components from the previous results using the same format? The following two formats do not work  
```
[velocity]
    family = LAGRANGE_VEC
    initial_from_file_var = velocity
 []

[velocity]
    family = LAGRANGE_VEC
    initial_from_file_var = velocity_x velocity_y
 []
```
Please let me know how I can get this to work. 

Thanks,

Anant

**Comment by GiudGiud** [comment]
What did you generate them with ? In the first simulation, was velocity a lagrange_vec?

In the Mesh block, did you remember to set use_for_exodus_restart = true?

**Comment by raj-anant** [comment]
The first simulation also had velocity as a lagrange_evc. 

I did not use the use_forexodus_restart = true in the mesh block in the previous simulation.

I tried using that in a fresh run, but it gave me an error: "unused parameter 'Mesh/use_for_exodus_restart'"

**Reply by GiudGiud** [reply]
use a FileMeshGenerator to load the mesh
The parameter might be mispelled too. This parameter must be used to restart from an exodus file

**Reply by raj-anant** [reply]
I used that tag as you suggested
```
[Mesh]
  [file_mesh]
     type = FileMeshGenerator
     file = melt_pool_old.e
     use_for_exodus_restart = true
   []
[]

[velocity]
    family = LAGRANGE_VEC
    initial_from_file_var = velocity
[]
```
The velocity was a LAGRANGE_VEC in the initial simulation as well. I still get the same error when trying to read the velocity:

Available variables: 
curvature
grad_ls_x
grad_ls_y
ls
p
temp
velocity_x
velocity_y
libMesh terminating:
Unable to locate variable named: velocity

What do you think?

**Reply by GiudGiud** [reply]
Looks like we dont support exodus restart of vector variables then
@cticenhour what did you use for restart in EM? for loading variables into other variables at initialization

**Reply by cticenhour** [reply]
I simply don't do restarts in the EM module right now because of this incompatibility. It's a pretty serious (and frustrating) gap, in my opinion. I am hopeful that the milestone work I am doing for the initial condition system this summer will help address this.

**Reply by raj-anant** [reply]
Is there any other way of reading the initial conditions for a variable from a .csv file or any other format? Or maybe you could give me some pointers as to how I could write some custom scripts inside MOOSE to read initial data from any file? I need to be able to read in randomly perturbed initial states for the problem I am working on.

**Reply by raj-anant** [reply]
ChatGPT suggested the following round about solution:
```
[ICs]
  [./velocity_x_ic]
    type = FunctionIC
    variable = velocity_x
    function = 'fromfile_x'
  [../]
  [./velocity_y_ic]
    type = FunctionIC
    variable = velocity_y
    function = 'fromfile_y'
  [../]
[]

[Functions]
  [./fromfile_x]
    type = ParsedFunction
    value = 'restartable_data_x'
  [../]
  [./fromfile_y]
    type = ParsedFunction
    value = 'restartable_data_y'
  [../]
[]

[RestartableData]
  [./restartable_data_x]
    type = ExodusReader
    file = 'previous_simulation.e'
    variable = 'velocity_x'
  [../]
  [./restartable_data_y]
    type = ExodusReader
    file = 'previous_simulation.e'
    variable = 'velocity_y'
  [../]
[]
```
Is something along these lines possible? I tried it but got an error: '[RestartableData]' does not have an associated "Action"

**Reply by cticenhour** [reply]
You could try using the [VectorFunctionIC](https://mooseframework.inl.gov/source/ics/VectorFunctionIC.html) object and define functions for each component given the requirements you have. Example input files using this are listed at the bottom of that page. You then might be able to define a suitable function from CSV for each by using the [PiecewiseConstantFromCSV](https://mooseframework.inl.gov/source/functions/PiecewiseConstantFromCSV.html) object. Note on the latter page a section containing links to other function objects that can take CSV data if this one isn't adequate.

**Reply by GiudGiud** [reply]
ChatGPT is making up syntax here. We dont have that one

**Comment by hugary1995** [comment]
Don't do variable reinitialization then... You can do `Problem/restart_file_base=xxxxx` or something like that (I don't recall the exact syntax).

**Reply by GiudGiud** [reply]
Instead of using the 'Exodus' restart, you should use the 'Checkpoint' restart. It s less flexible but it will support the vector variables
